src/classes/frUtil.cls

```diff
@@ -101,4 +81,64 @@ public class frUtil {
         }
     }
     
+    public static void logRelationshipError(Entity errObject, String recordId, Entity missingRelationship, String relationshipId) {
+        logRelationshipError(errObject, recordId, missingRelationship, relationshipId, null);
+    }
+    
+    public static void logRelationshipError(Entity errObject, String recordId, Entity missingRelationship, String relationshipId, String optionalError) {
+        String relationshipObject = getErrorObjectFromEnum(missingRelationship);
+        String error = 'Failed to find related record ' + 
+                 relationshipObject + ' with Funraise Id: ' + relationshipId;
+        if(optionalError != null) {
+            error += '. ' + optionalError;
+        }
+        logError(errObject, recordId, error);
+    }
+    
+    public static void logException(Entity errObject, String recordId, Exception ex) {
+        logError(errObject, recordId, 'Operation failed. Exception: '+ex.getMessage());
+    }
+    
+    public static void logError(Entity errObject, String recordId, String error) {
+		String funraiseObject = getErrorObjectFromEnum(errObject);
+        insert new Error__c(
+        	Error__c = error,
+            Funraise_Object__c = funraiseObject,
+            Funraise_Object_Id__c = recordId
```

Comment:
@adelorme82 The method you commented on will end up calling this one and it stores the record id in a special field now!  I didn't like that it was mixed in with the rest of the error, so the object type and record id are new fields for the error object, and then where you commented is still the old way of adding additional context to the error since in the relationship case there's ANOTHER record id that is interesting/relevant